src/tests/mfem_petsc_smoketest.cpp

```diff
@@ -0,0 +1,453 @@
+   //     system, applying any necessary transformations such as: parallel
+   //     assembly, eliminating boundary conditions, applying conforming
+   //     constraints for non-conforming AMR, static condensation, etc.
+   if (static_cond) { a->EnableStaticCondensation(); }
+   a->Assemble();
+
+   OperatorPtr A;
+   Vector B, X;
+   a->FormLinearSystem(ess_tdof_list, x, *b, A, X, B);
+
+   // 13. Solve the linear system A X = B.
+   //     If using MFEM with HYPRE
+   //       * With full assembly, use the BoomerAMG preconditioner from hypre.
+   //       * With partial assembly, use Jacobi smoothing, for now.
+   //     If using MFEM with PETSc
+   //       * With full assembly, use command line options or H2 matrix solver
+   //       * With partial assembly, wrap Jacobi smoothing, for now.
+   Solver *prec = NULL;
+   if (pa)
+   {
+      if (UsesTensorBasis(*fespace))
+      {
+         prec = new OperatorJacobiSmoother(*a, ess_tdof_list);
+      }
+   }
+   else
+   {
+      prec = new HypreBoomerAMG;
+   }
+
+   if (!use_petsc)
+   {
+      CGSolver *pcg = new CGSolver(MPI_COMM_WORLD);
+      if (prec) { pcg->SetPreconditioner(*prec); }
+      pcg->SetOperator(*A);
+      pcg->SetRelTol(1e-12);
+      pcg->SetMaxIter(200);
+      pcg->SetPrintLevel(1);
+      pcg->Mult(B, X);
+      delete pcg;
+   }
+   else
+   {
+      PetscPCGSolver *pcg;
+      // If petscrc_file has been given, we convert the HypreParMatrix to a
+      // PetscParMatrix; the user can then experiment with PETSc command line
+      // options unless forcewrap is true.
+      bool wrap = forcewrap ? true : (pa ? true : !strlen(petscrc_file));
+      if (wrap)
+      {
+         pcg = new PetscPCGSolver(MPI_COMM_WORLD);
+         pcg->SetOperator(*A);
+         if (useh2)
+         {
+            delete prec;
+            prec = new PetscH2Solver(*A.Ptr(),fespace);
+         }
+         else if (!pa) // We need to pass the preconditioner constructed from the HypreParMatrix
+         {
+            delete prec;
+            HypreParMatrix *hA = A.As<HypreParMatrix>();
+            prec = new HypreBoomerAMG(*hA);
+         }
+         if (prec) { pcg->SetPreconditioner(*prec); }
+      }
+      else // Not wrapping, pass the HypreParMatrix so that users can experiment with command line
+      {
+         HypreParMatrix *hA = A.As<HypreParMatrix>();
+         pcg = new PetscPCGSolver(*hA, false);
+         if (useh2)
+         {
+            delete prec;
+            prec = new PetscH2Solver(*hA,fespace);
+         }
+      }
+      pcg->iterative_mode = true; // iterative_mode is true by default with CGSolver
+      pcg->SetRelTol(1e-12);
+      pcg->SetAbsTol(1e-12);
+      pcg->SetMaxIter(200);
+      pcg->SetPrintLevel(1);
+
+      UserMonitor mymon(a,b);
+      if (visualization && petscmonitor)
+      {
+         pcg->SetMonitor(&mymon);
+         pcg->iterative_mode = true;
+         X.Randomize();
+      }
+      pcg->Mult(B, X);
```

Comment:
```suggestion
      pcg->Mult(B, X);
// SERAC_EDIT_BEGIN
      KSPConvergedReason reason;
      KSPGetConvergedReason(pcg, &reason);
      EXPECT_GT(reason, 0);
// SERAC_EDIT_END
```
We should probably check that we actually solved the system, given that the check is pretty free.